src/wallet/wallet.cpp

```diff
@@ -3140,22 +3140,18 @@ bool CWallet::TopUpKeyPool(unsigned int kpSize)
                 internal = true;
             }
 
-            if (!setInternalKeyPool.empty()) {
-                nEnd = *(setInternalKeyPool.rbegin()) + 1;
-            }
-            if (!setExternalKeyPool.empty()) {
-                nEnd = std::max(nEnd, *(setExternalKeyPool.rbegin()) + 1);
-            }
+            int64_t index = ++m_max_keypool_index;
+            assert(index >= 0); // How in the hell did you use so many keys?
```

Comment:
Code like this makes me go "wtf were the authors of this smoking": asserts like this are usually optimized out, so the code looks like it's saying that the author thought the signed overflow was kosher and that they thought it could happen here.